keps/sig-network/3698-multi-network-requirements/README.md

```diff
@@ -0,0 +1,352 @@
+that Namespace are automatically attached to.
+
+<p align="center">
+  <img src="mn-story-6.png?raw=true" alt="multi-network story 6 namespace networks profiles"/>
+</p>
+
+#### Story #7
+As a “Power User” with admin privileges I wish to have the ability to modify my
+Pod network namespace without any restrictions. I am aware that by doing this I
+might break established contracts for the kubernetes features.
+
+#### Story #8
+As a Virtual Machine -based compute platform provider that I run on top of
+Kubernetes and Kubevirt I need the ability to add/remove Network Interfaces to
+existing VMs without re-creating the VM. This would not be applicable for the
+Primary PodNetwork.
+
+<p align="center">
+  <img src="mn-story-8.png?raw=true" alt="multi-network story 8 network interface hot-plug"/>
+</p>
+
+#### Story #9
+As an infrastructure administrator I wish to support IPv6 SLAAC in the
+Kubernetes Pods. SLAAC's nature is to dynamically assign IP addresses and change
+them over time. Those should be reflected in Pod status. This would not be
+applicable for the Primary PodNetwork.
+
+### Requirements
+
+Below requirements are divided into phases. Each phase will produce a separate
+KEP with detailed design for specified set of requirements. Each of this KEP
+will have to take all of the requirements, from each phase, into consideration
+when created.
+
+#### Phase I (base API and reference in Pod)
+1. This effort shall not change the behavior of Today existing clusters
+2. We need an API object representing network, which a Pod, a Service or any
+other feature that wishes to support Multi-Network capability could reference
+   * We will introduce an PodNetwork API object that represents the existing 
+     infrastructure’s networking
+3. PodNetwork is decoupled from the infrastructure’s network
+definition/description created by the cluster operator. It functions as a handle
+4. PodNetwork shall not define any implementation specific parameters in its
+specification
+5. PodNetwork shall provide cluster generic options only
+6. PodNetwork can reference to implementation-specific parameters
+7. PodNetwork is the object Application developer will reference in their workloads
+8. Cluster Default PodNetwork is the PodNetwork the cluster has been created with
+9. Cluster Default PodNetwork cannot be changed or removed 
+10. Pods can reference one or more PodNetworks when trying to attach to them
+    * This list is explicit and all the PodNetworks that has to be available in
+      the Pod has to be listed
+11. Workloads access to PodNetwork is controlled via RBAC configuration
+12. When Pods does not specify any reference to PodNetwork it connects to Cluster
+Default PodNetwork (network the cluster has been created with)
+13. Pod shall be able to provide additional configuration on how it attaches to
+a PodNetwork
+14. Every Pod connected to specific PodNetwork must have connectivity within
```

Comment:
Do we want to define what "connectivity" means specifically.

We should state explicitly that we are probably never going to mandate a requirement (think K8s conformance) that two networks have or don't have connectivity to each other because it is out of scope of a given K8s cluster -- e.g. someone can always insert a connection there outside of the cluster infrastructure and we can't rely or enforce this easily.

Reply:
I dont think we specify what "connectivity" means in Today's Kubernetes network requirement neither. Additionally, PodNetwork's implementations might be done with different technologies which will make it hard to have one definition for it.

I have created separate requirement for the X-connectivity. keeping it formulated as a requirement.